In [3]:
import pandas
import seaborn
import matplotlib.pyplot as plt

url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/chipotle.tsv'

chipotle_data = pandas.read_csv(url, sep = '\t')

chipotle_data.head(20)

order_id  quantity                              item_name  \
0          1         1           Chips and Fresh Tomato Salsa   
1          1         1                                   Izze   
2          1         1                       Nantucket Nectar   
3          1         1  Chips and Tomatillo-Green Chili Salsa   
4          2         2                           Chicken Bowl   
5          3         1                           Chicken Bowl   
6          3         1                          Side of Chips   
7          4         1                          Steak Burrito   
8          4         1                       Steak Soft Tacos   
9          5         1                          Steak Burrito   
10         5         1                    Chips and Guacamole   
11         6         1                   Chicken Crispy Tacos   
12         6         1                     Chicken Soft Tacos   
13         7         1                           Chicken Bowl   
14         7         1                    Chips and Guacamole   
15         8         1  Chips and Tomatillo-Green Chili Salsa   
16         8         1                        Chicken Burrito   
17         9         1                        Chicken Burrito   
18         9         2                            Canned Soda   
19        10         1                           Chicken Bowl   

                                   choice_description item_price  
0                                                 NaN     $2.39   
1                                        [Clementine]     $3.39   
2                                             [Apple]     $3.39   
3                                                 NaN     $2.39   
4   [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5   [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                 NaN     $1.69   
7   [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8   [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9   [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25   
10                                                NaN     $4.45   
11  [Roasted Chili Corn Salsa, [Fajita Vegetables,...     $8.75   
12  [Roasted Chili Corn Salsa, [Rice, Black Beans,...     $8.75   
13  [Fresh Tomato Salsa, [Fajita Vegetables, Rice,...    $11.25   
14                                                NaN     $4.45   
15                                                NaN     $2.39   
16  [Tomatillo-Green Chili Salsa (Medium), [Pinto ...     $8.49   
17  [Fresh Tomato Salsa (Mild), [Black Beans, Rice...     $8.49   
18                                           [Sprite]     $2.18   
19  [Tomatillo Red Chili Salsa, [Fajita Vegetables...     $8.75

In [2]:
chipotle_data.describe()

order_id     quantity
count  4622.000000  4622.000000
mean    927.254868     1.075725
std     528.890796     0.410186
min       1.000000     1.000000
25%     477.250000     1.000000
50%     926.000000     1.000000
75%    1393.000000     1.000000
max    1834.000000    15.000000

Based on the method we used,
1. There were 4622 different rows in the data set.
2. On average, people ordered 1 item.
3. The minimum quantity of items ordered is 1.
4. There were 1834 orders in total.
5. The largest quantity of things ordered for only 1 kind of item was 15.
6. Less than 25 % of people ordered more than 1 item.

In [4]:
# listing all the columns and their data types.
chipotle_data.dtypes

order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

In [11]:
# figure out all the items that can be ordered
# how many times each ordered

group = chipotle_data.groupby('item_name').sum()
group.sort_values("quantity", ascending=False)

order_id  quantity  \
item_name                                                   
Chicken Bowl                             713926       761   
Chicken Burrito                          497303       591   
Chips and Guacamole                      449959       506   
Steak Burrito                            328437       386   
Canned Soft Drink                        304753       351   
Chips                                    208004       230   
Steak Bowl                               193752       221   
Bottled Water                            175944       211   
Chips and Fresh Tomato Salsa             100419       130   
Canned Soda                               76396       126   
Chicken Salad Bowl                       117104       123   
Chicken Soft Tacos                        98395       120   
Side of Chips                             84769       110   
Veggie Burrito                            80962        97   
Barbacoa Burrito                          74718        91   
Veggie Bowl                               72270        87   
Carnitas Bowl                             62742        71   
Barbacoa Bowl                             53972        66   
Carnitas Burrito                          51652        60   
Steak Soft Tacos                          50177        56   
6 Pack Soft Drink                         52322        55   
Chips and Tomatillo Red Chili Salsa       45556        50   
Chicken Crispy Tacos                      47382        50   
Chips and Tomatillo Green Chili Salsa     41170        45   
Carnitas Soft Tacos                       37673        40   
Steak Crispy Tacos                        35818        36   
Chips and Tomatillo-Green Chili Salsa     21826        33   
Steak Salad Bowl                          29590        31   
Nantucket Nectar                          21814        29   
Chips and Tomatillo-Red Chili Salsa       18344        25   
Barbacoa Soft Tacos                       18725        25   
Chips and Roasted Chili Corn Salsa        14999        23   
Izze                                      10587        20   
Veggie Salad Bowl                         17303        18   
Chips and Roasted Chili-Corn Salsa        17462        18   
Barbacoa Crispy Tacos                      5613        12   
Barbacoa Salad Bowl                        9708        10   
Chicken Salad                             10465         9   
Carnitas Crispy Tacos                      5868         8   
Veggie Soft Tacos                          5520         8   
Burrito                                    1550         6   
Veggie Salad                               7265         6   
Carnitas Salad Bowl                        6693         6   
Bowl                                        472         4   
Steak Salad                                2362         4   
Salad                                      1150         2   
Crispy Tacos                                434         2   
Chips and Mild Fresh Tomato Salsa           279         1   
Carnitas Salad                             1500         1   
Veggie Crispy Tacos                         668         1   

                                                                      choice_description  \
item_name                                                                                  
Chicken Bowl                           [Tomatillo-Red Chili Salsa (Hot), [Black Beans...   
Chicken Burrito                        [Tomatillo-Green Chili Salsa (Medium), [Pinto ...   
Chips and Guacamole                                                                    0   
Steak Burrito                          [Tomatillo Red Chili Salsa, [Fajita Vegetables...   
Canned Soft Drink                      [Coke][Sprite][Coke][Coke][Lemonade][Sprite][D...   
Chips                                                                                  0   
Steak Bowl                             [Fresh Tomato Salsa (Mild), [Black Beans, Rice...   
Bottled Water                         